pkg/node/manager/manager.go

```diff
@@ -734,6 +734,49 @@ func (m *manager) NodeUpdated(n nodeTypes.Node) {
 	m.ipsetMgr.AddToIPSet(ipset.CiliumNodeIPSetV4, ipset.INetFamily, v4Addrs...)
 	m.ipsetMgr.AddToIPSet(ipset.CiliumNodeIPSetV6, ipset.INet6Family, v6Addrs...)
 
+	if !n.IsLocal() {
```

Comment:
Why is it gated on whether the node is remote to insert pod CIDRs into the ipcache? We don't need the pod CIDRs in the ipcache for the local node?

Reply:
Ah I missed that it was answered here already: https://github.com/cilium/cilium/pull/37026#discussion_r1934378103

In that case, could we add a comment to the code to clarify why this is being done?